### PR TITLE
Assertion editor

### PR DESCRIPTION
**Pull request by rprimet**
Draft for discussion (WIP, not ready yet)

### CODE REVIEW

**Comment by rprimet** [comment]
@vincent-botbol @denismerigoux if that's OK with you we could merge this as a first step, then implement the 'add test' wizard and re-do an iteration on the assertion editor for composite structures?

File: src/defaults.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  switch (typ.kind) {
+    case 'TBool':
+      return { kind: 'Bool', value: false };
+    case 'TInt':
+    case 'TMoney':
+      return { kind: 'Integer', value: 0 };
+    case 'TRat':
+      return { kind: 'Decimal', value: 0 };
+    case 'TDate':
+      return { kind: 'Date', value: { year: 1970, month: 1, day: 1 } };
```
Comment (denismerigoux) [inline src/defaults.ts:14]: Default should be today's date no?

File: src/defaults.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    case 'TTuple':
+    case 'TStruct':
+    case 'TEnum':
+    case 'TOption':
+    case 'TArray':
+      throw new Error(
+        `Default value not implemented for compound type: ${typ.kind}`
+      );
+    default:
+      assertUnreachable(typ);
```
Comment (denismerigoux) [inline src/defaults.ts:26]: This could be implemented easily no?

**Review by denismerigoux — APPROVED** [review]
Ok for merging now for me!

**Pull request merged**